cores/esp8266/core_esp8266_si2c.cpp

```diff
@@ -714,8 +715,8 @@ void ICACHE_RAM_ATTR Twi::onSclChange(void)
     unsigned int scl;
 
     // Store bool return in int to reduce final code size.
-    sda = twi.SDA_READ();
-    scl = twi.SCL_READ();
+    sda = (GPI & (1 << twi.twi_sda)) != 0; // SDA_READ() fast, no time for a call
```

Comment:
Don't access the global twi instance from the class. Just access the twi_sda member directly.